core/enumerator/product_spec.rb

```diff
@@ -2,6 +2,29 @@
 
 ruby_version_is "3.2" do
   describe "Enumerator.product" do
+    it "creates a finite enumerator" do
+      enum = Enumerator.product(1..2, ["A", "B"])
+      enum.to_a.should == [[1, "A"], [1, "B"], [2, "A"], [2, "B"]]
+    end
+
+    it "accepts a list of enumerators of any length" do
+      enum = Enumerator.product(1.., ["A"], ["B"])
+      enum2 = Enumerator.product(2.., ["A"], ["B"], ["C"])
+
+      enum.take(2).should == [[1, "A", "B"], [2, "A", "B"]]
+      enum2.take(2).should == [[2, "A", "B", "C"], [3, "A", "B", "C"]]
```

Comment:
**minor:** I would avoid infinite enumerators, and test them only in a dedicated test case as far as it's a corner case but it requires additional calls of `take` method, that makes the test a bit more complex.

```suggestion
      enum = Enumerator.product(1..2, ["A"], ["B"])
      enum.to_a.should == [[1, "A", "B"], [2, "A", "B"]]
      
      enum = Enumerator.product(2..3, ["A"], ["B"], ["C"])
      enum.to_a.should == [[2, "A", "B", "C"], [3, "A", "B", "C"]]
```